R/spatial_clustering_cv.R

```diff
@@ -32,17 +39,67 @@
 #' pp. 5372-5375, doi: 10.1109/IGARSS.2012.6352393.
 #'
 #' @examples
-#' data(ames, package = "modeldata")
-#' spatial_clustering_cv(ames, coords = c(Latitude, Longitude), v = 5)
+#' data(Smithsonian, package = "modeldata")
+#' spatial_clustering_cv(Smithsonian, coords = c(latitude, longitude), v = 5)
+#'
+#' # When providing sf objects, coords are inferred automatically
+#' smithsonian_sf <- sf::st_as_sf(Smithsonian,
+#'                         coords = c("longitude", "latitude"),
+#'                         crs = 4326) # WGS84
+#' spatial_clustering_cv(smithsonian_sf, v = 5)
+#'
+#' @rdname spatial_clustering_cv
 #' @export
-spatial_clustering_cv <- function(data, coords, v = 10, ...) {
-  coords <- tidyselect::eval_select(rlang::enquo(coords), data = data)
+spatial_clustering_cv <- function(data, coords, v = 10, fun = c("kmeans", "hclust"), ...) {
+  UseMethod("spatial_clustering_cv", data)
+}
 
+#' @rdname spatial_clustering_cv
+#' @export
+spatial_clustering_cv.default <- function(data, coords, v = 10, fun = c("kmeans", "hclust"), ...) {
+  fun <- match.arg(fun)
```

Comment:
`rlang::arg_match()` is a bit stricter (no partial matching) and the error messages obey tidyverse standards so I would use that here